src/v/cluster/nested_partition_probe.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "metrics/metrics.h"
+#include "model/fundamental.h"
+
+namespace cluster {
+
+class nested_partition_probe {
```

Comment:
nit: not sure if it's just me, but when i think of something being nested i think of it being recursively nested. Maybe this could be some sub_partition_probe or specialized_partition_probe?

Also it'd be great to add a comment about the motivation of this class, and its expected relationship to partition_probe.